File: JULY-2025/01-07-2025/p3330.ex
defmodule Solution do
  @spec possible_string_count(word :: String.t) :: integer
  def possible_string_count(word) do
    for <<ch::8 <- word>>, reduce: {nil, 1} do
      {prev_ch, res} ->
        if prev_ch == ch, do: {ch, res + 1}, else: {ch, res}
    end |> elem(1)
  end
end

File: JULY-2025/01-07-2025/p3330.rs
impl Solution {
  pub fn possible_string_count(word: String) -> i32 {
    let mut res = 1;
    let mut prev_c = 'A';

    for c in word.chars() {
      if c == prev_c {
        res += 1;
      } else {
        prev_c = c;
      }
    }

    res
  }
}